MekHQ/unittests/mekhq/campaign/parts/RefitTest.java

```diff
@@ -0,0 +1,224 @@
+        when(mockCampaign.getWarehouse()).thenReturn(mockWarehouse);
+
+        // Create the original entity backing the unit
+        Entity oldEntity = UnitTestUtilities.getLocustLCT1V();
+        IPlayer mockPlayer = mock(IPlayer.class);
+        when(mockPlayer.getName()).thenReturn("Test Player");
+        oldEntity.setOwner(mockPlayer);
+
+        // Create the entity we're going to refit to
+        Entity newEntity = UnitTestUtilities.getLocustLCT1E();
+
+        // Create the unit which will be refit
+        Unit oldUnit = new Unit(oldEntity, mockCampaign);
+        oldUnit.initializeParts(false);
+
+        // Create the Refit
+        Refit refit = new Refit(oldUnit, newEntity, false, false);
+        assertEquals(mockCampaign, refit.getCampaign());
+
+        // Should be old parts...
+        assertFalse(refit.getOldUnitParts().isEmpty());
+
+        // ...and new parts.
+        assertFalse(refit.getNewUnitParts().isEmpty());
+
+        // ...and we'll need to buy some parts
+        assertFalse(refit.getShoppingList().isEmpty());
+    }
+
+    @Test
+    public void locust1Vto1ETest() {
+        Campaign mockCampaign = mock(Campaign.class);
+        Warehouse mockWarehouse = mock(Warehouse.class);
+        when(mockCampaign.getWarehouse()).thenReturn(mockWarehouse);
+
+        // Create the original entity backing the unit
+        Entity oldEntity = UnitTestUtilities.getLocustLCT1V();
+        IPlayer mockPlayer = mock(IPlayer.class);
+        when(mockPlayer.getName()).thenReturn("Test Player");
+        oldEntity.setOwner(mockPlayer);
+
+        // Create the entity we're going to refit to
+        Entity newEntity = UnitTestUtilities.getLocustLCT1E();
+
+        // Create the unit which will be refit
+        Unit oldUnit = new Unit(oldEntity, mockCampaign);
+        oldUnit.setId(UUID.randomUUID());
+        oldUnit.initializeParts(false);
+
+        // Create the Refit
+        Refit refit = new Refit(oldUnit, newEntity, false, false);
+        assertEquals(mockCampaign, refit.getCampaign());
+
+        //
+        // Locust 1V to 1E Class D refit steps (in no particular order):
+        //     1. Remove excess Machine Gun (LA) [120 mins]
+        //     2. Remove excess Machine Gun (RA) [120 mins]
+        //     3. Remove Machine Gun Ammo Bin (CT) [120 mins]
+        //     4. Move Medium Laser (CT) to (RA) [120 mins]
+        //     5. Add Medium Laser to (LA) [120 mins]
+        //     6. Add Small Laser to (RA) [120 mins]
+        //     7. Add Small Laser to (LA) [120 mins]
+        //
+        // Everything else is the same.
+        //
+
+        // TODO: confirm this
+        assertEquals(Refit.CLASS_D, refit.getRefitClass());
+
+        // Time?
+        //     + 3 removals @ 120 mins ea
+        //     + 1 move @ 120 mins ea
+        //     + 3 adds @ 120 mins ea
+        //     x 3 (Class D)
+        assertEquals((120.0 * 7.0) * 3.0, refit.getActualTime(), 0.1);
```

Comment:
What's the .1, is that like a tolerance?

Reply:
Yes. That's to deal with floating point errors when comparing floats and doubles.